### PR TITLE
Update deprecated flags and commands to print out a message about v1.0

### PR DESCRIPTION
**Pull request by bufdev**

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]
👌🏻

**Review by doriable — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,49 +2,60 @@
...
-- Removed the `buf beta mod export` command in favor of `buf export`.
-- Removed the `buf beta mod init` command in favor of `buf config init`.
-- Removed the `buf beta mod update` command in favor of `buf mod update`.
-- Removed the `buf beta mod clear-cache` command in favor of `buf mod clear-cache`.
-- Removed the `buf beta push` command in favor of `buf mod push`.
+This is our first v1.0 release candidate. This release largely concentrates on erroring for
+already-deprecated commands and flags.
+
+At Buf, we take compatibility very seriously. When we say v1.0, we mean it - we hope `buf` will be
+stable on v1 for the next decade, and if there is something we want to change, it is our responsibility to
```
Comment (amckinney) [inline CHANGELOG.md:9]: nit: We use `v1.0` and `v1` - we should probably just use one form. Maybe `v1` so that it suggests it includes all minors.

**Review by amckinney — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bufdev) [reply CHANGELOG.md:9]: Eh I thought about that but I kind of did this on purpose - v1.0 has more of a ring to it, but v1 is more correct in the stability context. Idk.

Comment (amckinney) [reply CHANGELOG.md:9]: Yea that's fair, feel free to leave as-is.

**Pull request merged**